src/cls/rbd/cls_rbd.cc

```diff
@@ -4421,6 +4431,337 @@ int group_dir_remove(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
   return 0;
 }
 
+/**
+ * Add an image to the consistency group.
+ *
+ * Input:
+ * @param image_id (std::string)
+ * @param pool_id (int64_t)
+ *
+ * Output:
+ * @return 0 on success, negative error code on failure
+ */
+int group_image_add(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
```

Comment:
Minor: could change this to a set state method and remove the need for the extra _clean method.  Also needed if DETACHING state required.